projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -165,9 +165,72 @@ def initialize_job(context: JobContext) -> None:
 
 
 class IIngesterPlugin:
+    """
+    Interface representing the APIs that can be used when creating Ingester
+    Plugins. Plugins implement one of the methods, depending on their needs.
+    If more complex processing is required, plugins can be chained
+    together.
+
+    The lifecycles of the three methods are as follows:
+
+    ingest_payload() - plugins implementing this method are intended
+                       to do actual data ingestion. Versatile Data Kit will
+                       call this method when the payload that data engineers
+                       have passed is ready to be ingested.
+
+    process_payload() - plugins implementing this method do not aim at
+                        ingesting the payload, rather to do some processing
+                        of the payload before it is ingested.
+
+    post_ingest_process() - plugins implementing this method aim at doing
+                            some metadata processing after the payload is
+                            ingested. The main use-case is telemetry data
+                            collection.
+                            NOTE: This method can be used only by plugins
+                            that implement ingest_payload().
+
+    Example plugin chaining workflows:
+
+    Workflow 1
+    ----------
+    _____________________      _________________________      ______________
+    |      Data Job     |      |    vdk-ingest-http    |      |            |
+    |-------------------|      |-----------------------|      | Data Lake/ |
+    |                   |      |                       |      | Warehouse/ |
+    |send_object_       | ---> |   ingest_payload()    | ---> | Third-party|
+    |    for_ingestion()|      |                       |      | endpoint   |
+    |___________________|      |_______________________|      |____________|
+
+
+    Workflow 2
+    ----------                                                  ______________
+                                                                |            |
+    _____________________      ___________________________      | Data Lake/ |
+    |      Data Job     |      |     vdk-ingest-http     |      | Warehouse/ |
+    |-------------------|      |-------------------------|      | Third-party|
+    |                   |      | -- ingest_payload()     | ---> | endpoint   |
+    |send_object_       | ---> | |                       |      |____________|
+    |    for_ingestion()|      | |                       |      ______________
+    |                   |      | -->post_ingest_process()| ---> |            |
```

Comment:
Most likely it would be a separate plugin that implements telemetry. 

For example, we can send telemetry to apache Atlas using vdk-atlas plugin or opentelemetry using vdk-opentelemetry plugin or some internal telemetry using vdk-http-plugin (which is the case internally in VMware).

Reply:
See my comment in the thread above. Thanks